### ISSUE TITLE
Intermittent erroneous error about calling a child command before running a parent command

### ISSUE THREAD

**Issue by jpage-godaddy** [cypress-io/cypress#30522]
### Current behavior

I'm occasionally getting an error in my test suite. I have a `beforeEach` hook like this:

```javascript
    beforeEach(function () {
      cy.intercept('GET', `/some/api/endpoint`, { statusCode: 200, body: [] })
        .as('getAppSettings');
    });
```

...and sometimes I get a failure like this:

> The following error originated from your test code, not from Cypress. It was caused by an unhandled promise rejection.
>
> \> Oops, it looks like you are trying to call a child command before running a parent command.
> 
> You wrote code that looks like this:
>
> cy.as("getAppSettings")
>
> A child command must be chained after a parent because it operates on a previous subject.

Clearly something else is going on because I am not doing what the error says I'm doing. It seems like an unhandled exception occurring in the prior test causing an error to be misidentified. If I skip the prior test, this does not occur.

### Desired behavior

Cypress should show the actual unhandled exception

### Test code to reproduce

I cannot reproduce in a minimal enough way to share the full code.

### Cypress Version

13.15.1

### Node version

20.18.0

### Operating System

Ubuntu 22.04.4 LTS

### Debug Logs

```shell
I didn't see any debug output relating to this issue.
```

### Other

I think the uncaught exception is due to an error being thrown in the browser _after_ a previous test has completed, and that is being caught by the _next_ test and erroneously associated with the currently executing command.

**Comment by jpage-godaddy** [comment]
Actually, this may be even simpler. It looks like in this test suite, someone erroneously made their test block an async function which I know is incorrect. Changing that fixes the issue. Maybe Cypress can detect if a `Promise` is returned when running a test and tell the user that they should not be using an an async function.

**Comment by tolkamps1** [comment]
I"m having the exact same thing happen with my tests, but it seems to be unreliable to reproduced. I'll run the suite 5 times with no issues, but then the next time I'll get the same error. This makes sense if it is an `async` function timing causing it, I'll take a look to see if I can track it down. 
Seconding the request for Cypress to detect this and provide a more informed error/warning.

**Comment by SanderKleykens** [comment]
We're seeing the same issue with Cypress v14.3.2. 
As far as I can tell, we have no tests that are async. I do see async functions (routeHandlers) being passed to cy.intercept, but as far as I can understand from the jsdocs on HttpRequestInterceptor, that should be ok.

**Comment by marktnoonan** [comment]
I have [an internal Cypress reproduction](https://cloud.cypress.io/projects/d9sdrd/runs/9413/test-results/46d201cf-84f8-4f85-b7ec-c649861171c7?actions=%5B%5D&browsers=%5B%5D&groups=%5B%5D&isFlaky=%5B%5D&modificationDateRange=%7B%22startDate%22%3A%221970-01-01%22%2C%22endDate%22%3A%222038-01-19%22%7D&orderBy=EXECUTION_ORDER&oses=%5B%5D&specs=%5B%5D&statuses=%5B%5D&testingTypesEnum=%5B%5D) that we can use to look into this further.

Version 15.5.0

Seems that this is rare but now that we have an example it should help.

**Comment by ryanjwilke** [comment]
Yes, this was definitely consistently failing locally and in CI. Effectively it was saying that code like this:
```
cy.intercept('{url}', '').as('alias')
```

...it was failing with an error that the `as` was running before `intercept` for some reason. I had to turn this line off in order to get past this issue.
